solver/src/driver.rs

```diff
@@ -80,11 +118,14 @@ impl Driver {
         tracing::debug!("got {} AMMs", amms.len());
 
         let liquidity: Vec<Liquidity> = limit_orders
+            .clone()
```

Comment:
nanonit: if you move `self.collect_estimated_prices(&limit_orders).await?;` above you shouldn't need to clone.